mysql-test/suite/rpl/r/rpl_show_slave_hosts.result

```diff
@@ -129,13 +129,114 @@ Server_id	Host	Port	Master_id	Gtid_State_Sent	Gtid_State_Ack
+# since IO thread of replicas is ON ?
+# Currently we observed the same `Gtid_State_Sent=Gtid_State_Ack=0-1-5`
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Gtid_State_Sent	Gtid_State_Ack
+3	slave2	SLAVE_PORT	1	0-1-5	0-1-5
+2	localhost	SLAVE_PORT	1	0-1-5	0-1-5
+connection slave;
+STOP SLAVE IO_THREAD;
+include/wait_for_slave_io_to_stop.inc
+show status like 'rpl_semi_sync_slave%';
+Variable_name	Value
+Rpl_semi_sync_slave_send_ack	5
+Rpl_semi_sync_slave_status	OFF
+show variables like 'rpl_semi_sync_slave_enabled';
+Variable_name	Value
+rpl_semi_sync_slave_enabled	ON
+connection master;
+show status like "Slaves_connected";
+Variable_name	Value
+Slaves_connected	1
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	1
+show variables like 'rpl_semi_sync_slave_enabled';
+Variable_name	Value
+rpl_semi_sync_slave_enabled	OFF
+# Q2: Here we don't get any row for server_id=2, but we should? we get only `Gtid_State_Sent= Gtid_State_Ack=0-1-5` for server_id=3
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Gtid_State_Sent	Gtid_State_Ack
+3	slave2	SLAVE_PORT	1	0-1-5	0-1-5
+# Write new event when IO thread is stopped
+INSERT INTO t VALUES (505);
+# Writing new events will affect only on replica server_id=3 Gtid_State_Sent= Gtid_State_Ack=0-1-6`
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Gtid_State_Sent	Gtid_State_Ack
+3	slave2	SLAVE_PORT	1	0-1-6	0-1-6
+connection slave;
+START SLAVE IO_THREAD;
+include/wait_for_slave_io_to_start.inc
+# Q*: Stoping and starting the new IO thread will restart slave sent ack count, see below.
+# So we basically want this to be recorded in edge case, right ? 
+# And that is rpl_semi_sync_master_timeout=0 and rpl_semi_sync_master_enabled=TRUE
+# Can we use this as patch instead of injecting the DBUG_SYNC?
+# We are expecting here `Rpl_semi_sync_slave_send_ack == 4` 
+show status like 'rpl_semi_sync_slave%';
+Variable_name	Value
+Rpl_semi_sync_slave_send_ack	0
+Rpl_semi_sync_slave_status	ON
+show variables like 'rpl_semi_sync_slave_enabled';
+Variable_name	Value
+rpl_semi_sync_slave_enabled	ON
+connection master;
+show status like "Slaves_connected";
+Variable_name	Value
+Slaves_connected	2
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	2
+# Q3: here we get  `Gtid_State_Sent= 0-1-5 != Gtid_State_Ack=''` (but Gtid_State_Ack should be 0-1-4 right?)
+# So should we get for server_id=2 `Gtid_State_Sent= 0-1-6 != Gtid_State_Ack=0-1-4` ?
+# Note if there were no new events here we woud get empty `Gtid_State_Sent= Gtid_State_Ack=''` for server_id=2, since server_id catched that event.
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Gtid_State_Sent	Gtid_State_Ack
+2	localhost	SLAVE_PORT	1	0-1-6	
+3	slave2	SLAVE_PORT	1	0-1-6	0-1-6
+connection slave;
+# Q4: here we get sync good with server_id=3, `Gtid_State_Sent= Gtid_State_Ack=0-1-6` for server_id=2?
```

Comment:
Good question, as this now deals with the active transaction logic.. Because there were no new transactions, this would present as the last transaction that the replica had sent an `ACK` for. I see that being potentially misleading to users when the replica does catch up, yet presents an old `Gtid_state_ack`. I see two options, either document the behavior (my preference), or we'd add additional `if (!timeout)` behavior to sometimes allow for "special" `ACK`s to non-active transactions.. @andrelkin what do you think?